internal/controllers/resourceslice/slice.go

```diff
@@ -0,0 +1,197 @@
+package resourceslice
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	apiv1 "github.com/Azure/eno/api/v1"
+	"github.com/Azure/eno/internal/manager"
+	"github.com/go-logr/logr"
+)
+
+// sliceController manages the lifecycle of resource slices in the context of their owning composition.
+// This consists of aggregating their status into the composition, and replacing missing slices.
+type sliceController struct {
+	client client.Client
+}
+
+func NewController(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&apiv1.Composition{}).
+		Owns(&apiv1.ResourceSlice{}).
+		WithLogConstructor(manager.NewLogConstructor(mgr, "sliceController")).
+		Complete(&sliceController{client: mgr.GetClient()})
+}
+
+func (s *sliceController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := logr.FromContextOrDiscard(ctx)
+
+	comp := &apiv1.Composition{}
+	err := s.client.Get(ctx, req.NamespacedName, comp)
+	if err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(fmt.Errorf("getting composition: %w", err))
```

Comment:
nit: Comment stating that deletion is handled by a different controller and some reasoning around why?